connector/oidc/oidc.go

```diff
@@ -189,7 +190,7 @@ func (c *oidcConnector) Close() error {
 }
 
 func (c *oidcConnector) LoginURL(s connector.Scopes, callbackURL, state string) (string, error) {
-	if c.redirectURI != callbackURL {
+	if c.redirectURI != callbackURL && os.Getenv("RELAX_ISSUER_CALLBACK_URL_EQUALITY_CHECK") != "true" {
```

Comment:
My 2c is that this could be a config option like `insecureEnableGroups` - the dex codebase tends to favour injecting config via these structs over reading an env var directly.